sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java

```diff
@@ -7825,7 +7827,8 @@ private boolean useThinClientStoreModel(RxDocumentServiceRequest request) {
 
         return operationType.isPointOperation()
                     || operationType == OperationType.Query
-                    || operationType == OperationType.Batch;
+                    || operationType == OperationType.Batch
+                    || ChangeFeedStateV1.isChangeFeedRequest(request) && !ChangeFeedStateV1.isAllVersionsAndDeletesChangeFeedMode(request);
```

Comment:
hmm, there is a `isChangeFeedRequest()` in RxDocumentServiceRequest class, maybe that is a better place to also add `isAllVersionsAndDeletesChangeFeedMode`